diff --git a/docker/Dockerfile_snap b/docker/Dockerfile_snap
index b8cab8dcd..c1d9048be 100644
--- a/docker/Dockerfile_snap
+++ b/docker/Dockerfile_snap
@@ -1,5 +1,5 @@
 ARG RISK=edge
-ARG UBUNTU=xenial
+ARG UBUNTU=bionic
 
 FROM ubuntu:$UBUNTU as builder
 ARG RISK
@@ -20,11 +20,11 @@ RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/sna
 RUN mkdir -p /snap/core
 RUN unsquashfs -d /snap/core/current core.snap
 
-# Grab the core18 snap (which snapcraft uses as a base) from the stable channel
+# Grab the core20 snap (which snapcraft uses as a base) from the stable channel
 # and unpack it in the proper place.
-RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core18' | jq '.download_url' -r) --output core18.snap
-RUN mkdir -p /snap/core18
-RUN unsquashfs -d /snap/core18/current core18.snap
+RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core20' | jq '.download_url' -r) --output core20.snap
+RUN mkdir -p /snap/core20
+RUN unsquashfs -d /snap/core20/current core20.snap
 
 # Grab the snapcraft snap from the $RISK channel and unpack it in the proper
 # place.
@@ -44,7 +44,7 @@ RUN chmod +x /snap/bin/snapcraft
 # time so they can be cached.
 FROM ubuntu:$UBUNTU
 COPY --from=builder /snap/core /snap/core
-COPY --from=builder /snap/core18 /snap/core18
+COPY --from=builder /snap/core20 /snap/core20
 COPY --from=builder /snap/snapcraft /snap/snapcraft
 COPY --from=builder /snap/bin/snapcraft /snap/bin/snapcraft
 
@@ -58,4 +58,8 @@ ENV LC_ALL="en_US.UTF-8"
 ENV PATH="/snap/bin:$PATH"
 ENV SNAP="/snap/snapcraft/current"
 ENV SNAP_NAME="snapcraft"
-ENV SNAP_ARCH="amd64"
\ No newline at end of file
+ENV SNAP_ARCH="amd64"
+
+ADD scripts/build-package-snap.sh /opt/build-package-snap.sh
+
+CMD ["/opt/build-package-snap.sh"]
